### PR TITLE
[DATA] Add TESS light curves

### PR DESCRIPTION
**Pull request by jeraud**
Include TESS data, still a draft. More to be added soon

### CODE REVIEW

**Comment by jeraud** [comment]
Still need to solve a couple of bugs but should be fixed soon.

File: scripts/tess/build_parent_sample.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import healpy as hp
+from astropy.units import cds
+from quality import TESSQualityFlags
+
+
+# Breakdown of the different TESS pipelines, each one will be stored as a subdataset
+PIPELINES = ['spoc  ']
+
+
+#def selection_fn(catalog):
```
Comment (helenqu) [inline scripts/tess/build_parent_sample.py:18]: remove commented out code

File: scripts/tess/build_parent_sample.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+def processing_fn(args):
+    """ Parallel processing function reading all requested light curves from one sector.
+    """
+    filename, object_id = args
+
+    with fits.open(filename, mode='readonly', memmap=True) as hdu:
+        # The header of hdu[0] contains the following information:
+        # Page 34, Section 5 - https://archive.stsci.edu/files/live/sites/mast/files/home/missions-and-data/active-missions/tess/_documents/EXP-TESS-ARC-ICD-TM-0014-Rev-F.pdf
+
+        telescope = hdu[0].header.get('TELESCOP')
```
Comment (helenqu) [inline scripts/tess/build_parent_sample.py:41]: is this a typo? TELESCOP

File: scripts/tess/build_parent_sample.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        cat_pipeline = catalog
+        #cat_pipeline = catalog[catalog['PIPELINE'] == pipeline]
+        
+        cat_pipeline = cat_pipeline.group_by(['healpix'])
+    
+        map_args = []
+        for group in cat_pipeline.groups:
+            # Create a filename for the group
+            group_filename = os.path.join(args.output_dir, '{}/healpix={}/001-of-001.hdf5'.format(pipeline.strip(), group['healpix'][0]))
+            if args.tiny:
```
Comment (helenqu) [inline scripts/tess/build_parent_sample.py:173]: don't need this if statement, args.tiny will resolve to false if the --tiny flag is not set

File: scripts/tess/tess.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    def _info(self):
+        """Defines the features available in this dataset."""
+        # Starting with all features common to time series datasets
+        features = {
+            "lightcurve": Sequence({
+                "time":  Value(dtype="float32"),
+                "flux": Value(dtype="float32"),
+                "flux_err": Value(dtype="float32"),
+                #"quality_mask": Value(dtype="int32"),
+                #"band": Value('string'),
```
Comment (helenqu) [inline scripts/tess/tess.py:93]: did you intentionally comment out these quantities?

File: scripts/tess/tess.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+_VERSION = "0.0.1"
+
+# Full list of features available here:
+# https://archive.stsci.edu/files/live/sites/mast/files/home/missions-and-data/active-missions/tess/_documents/EXP-TESS-ARC-ICD-TM-0014-Rev-F.pdf
+# https://cdsarc.cds.unistra.fr/ftp/IV/39/ReadMe
+#TODO: add additional features from the TIC catalog
+
+_FLOAT_FEATURES = [
+    "RA",
```
Comment (helenqu) [inline scripts/tess/tess.py:51]: no DEC? also is any other metadata, like object type available?

**Review by helenqu — COMMENTED** [review]
there's a lot of commented out code and TODOs - if you intended to delete them, please do!

File: scripts/tess/build_parent_sample.py (hunk 3 above)
Comment (jeraud) [reply scripts/tess/build_parent_sample.py]: Solved, thanks!

File: scripts/tess/tess.py (hunk 4 above)
Comment (jeraud) [reply scripts/tess/tess.py]: Yes! Directly subtracted the quality mask from the start to make things simpler.

File: scripts/tess/tess.py (hunk 5 above)
Comment (jeraud) [reply scripts/tess/tess.py:51]: I've added RA/DEC, it's also available directly as part of the light curve files (hadn't decided on the best way yet to include it). Object types are not available in one catalog unfortunately

**Review by EiffL — CHANGES_REQUESTED** [review]
This looks good to me @jeraud, and it works on my local  machine as well, could you just remove any commented code in `build_parent_sample.py` that is not needed? to make things a little bit cleaner.
After that I think we can merge,.

**Comment by jeraud** [comment]
Perfect! Removed the commented code in `build_parent_sample.py` and `mast_s3_transfer.py` now.

**Review by EiffL — APPROVED** [review]
LGTM! Thanks @jeraud !

**Pull request merged**